packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -516,16 +517,25 @@ class IrisGridTableModel extends IrisGridModel {
   get columns() {
     return this.getColumnsWithLayoutHints(
       this.table.columns,
-      this.table.layoutHints
+      this.layoutHints,
+      this.frozenColumns
     );
   }
 
+  get frozenColumns() {
+    if (this.userFrozenColumns && this.userFrozenColumns.length) {
```

Comment:
Should just be checking `this.userFrozenColumns != null`. Again, two cases - `[]` for when they explicitly unfreeze all columns from the UI, and `null` when they haven't frozen/unfrozen anything (only layoutHints are set)